geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/SimpleRasterIngest.scala

```diff
@@ -67,28 +69,47 @@ class SimpleRasterIngest(config: Map[String, Option[String]], cs: AccumuloCovera
   lazy val fileType = config(IngestRasterParams.FORMAT).get
   lazy val rasterName = config(IngestRasterParams.RASTER_NAME).get
   lazy val visibilities = config(IngestRasterParams.VISIBILITIES).get
-
+  lazy val parLevel = config(IngestRasterParams.PARLEVEL).get.toInt
 
   val df = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
 
   def runIngestTask() = Try {
-    val file = new File(path)
-    val ingestTime = config(IngestRasterParams.TIME).map(df.parseDateTime).getOrElse(new DateTime(DateTimeZone.UTC))
+    val fileOrDir = new File(path)
+    val ingestTime = config(IngestRasterParams.TIME).map(df.parseDateTime(_)).getOrElse(new DateTime(DateTimeZone.UTC))
 
-    val rasterReader = getReader(file, fileType)
-    val rasterGrid: GridCoverage2D = rasterReader.read(null)
+    val files =
+      (if (fileOrDir.isDirectory)
+         fileOrDir.listFiles(new FilenameFilter() {
+           override def accept(dir: File, name: String) =
+             getFileExtension(name).endsWith(fileType)
+         })
+       else Array(fileOrDir)).par
 
-    val envelope = rasterGrid.getEnvelope2D
+    files.tasksupport = new ForkJoinTaskSupport(new scala.concurrent.forkjoin.ForkJoinPool(parLevel))
+    files.foreach { file =>
```

Comment:
(Picky) For readability, I'd suggest making this lamdba a separate function which takes a file and persists.